libsolutil/CommonData.h

```diff
@@ -232,6 +232,17 @@ std::set<K> keys(std::map<K, V> const& _map)
 	return applyMap(_map, [](auto const& _elem) { return _elem.first; }, std::set<K>{});
 }
 
+/// @returns a pointer to the entry of @a _map at @a _key, if there is one, and nullptr otherwise.
+template<typename MapType, typename KeyType>
+auto fetchValue(MapType&& _map, KeyType const& _key)
```

Comment:
What we'd need would be ``optional<ValueType&>`` - but unfortunately ``std::optional`` doesn't work with reference types (I think that's actually a standard defect report IIRC...)... the closest thing to it is just a pointer.
An actual ``optional<ValueType>`` (no reference) would always require a copy, so that's best avoided, so the plain pointer is probably the best we can do...

Reply:
This function can be used all over the place throughout the compiler - I just didn't want to burden this PR with all that yet :-). There's 612 instances of ``.count(...)`` in the code-base - I'd estimate around half of those only use it as guard for accessing and will move to ``fetchValue``. A good part of those won't be stuff you'd want to copy.